### PR TITLE
Fix some external url formatting.

### PR DESCRIPTION
**Pull request by mklimasz**
Addresses some parts of #3630.

External urls in parameters sections are not rendering correctly (i.e. http://docs.allennlp.org/master/api/modules/stacked_alternating_lstm/). I have used reference style links (https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) to avoid that.

Also fixed some broken links in "regular" places.

### CODE REVIEW

File: allennlp/modules/stacked_alternating_lstm.py [hunk 1]
```diff
@@ -16,8 +16,10 @@ class StackedAlternatingLstm(torch.nn.Module):
     """
     A stacked LSTM with LSTM layers which alternate between going forwards over
     the sequence and going backwards. This implementation is based on the
-    description in [Deep Semantic Role Labelling - What works and what's next]
-    (https://homes.cs.washington.edu/~luheng/files/acl2017_hllz.pdf).
+    description in [Deep Semantic Role Labelling - What works and what's next][0].
+
+    [0]: https://homes.cs.washington.edu/~luheng/files/acl2017_hllz.pdf
```
Comment (dirkgr) [inline allennlp/modules/stacked_alternating_lstm.py:21]: Can you update this link to point to https://www.aclweb.org/anthology/P17-1044.pdf? The current one is dead.

Comment (mklimasz) [reply allennlp/modules/stacked_alternating_lstm.py]: Done, but doesn't it mean that script for url checking is not working (scripts/check_links.py) - masking it is worth making an issue?

Comment (dirkgr) [reply allennlp/modules/stacked_alternating_lstm.py]: Hm, that script actually fails when I run it, but due to a different error. If it's something small, I'll just fix it right here.

Comment (dirkgr) [reply allennlp/modules/stacked_alternating_lstm.py]: It fails because of other cases of the same problem. I fixed that just now, and added a commit.

Comment (dirkgr) [reply allennlp/modules/stacked_alternating_lstm.py]: Now, the `check_links.py` script fails because GitHub is sending out invalid headers. The `urllib` people think they can [wage a Twitter war to get it fixed on the GitHub end](https://github.com/urllib3/urllib3/issues/1822). Best not to get involved.

**Comment by dirkgr** [comment]
The stupid linter is going to be the end of me.

File: allennlp/modules/stacked_alternating_lstm.py (hunk 1 above)
Comment (mklimasz) [reply allennlp/modules/stacked_alternating_lstm.py]: Wow, that's very optimistic I guess.

Comment (pquentin) [reply allennlp/modules/stacked_alternating_lstm.py]: > Now, the `check_links.py` script fails because GitHub is sending out invalid headers. The `urllib` people think they can [wage a Twitter war to get it fixed on the GitHub end](https://github.com/urllib3/urllib3/issues/1822). Best not to get involved.

For what it's worth the person that suggested taking it to Twitter is not a maintainer, and never mentioned a war, just asked for a fix. Looks like it worked according to @andriyor.

Thanks for AllenNLP!

Comment (dirkgr) [reply allennlp/modules/stacked_alternating_lstm.py]: Looks like I should give GitHub (and @ssbarnea) more credit :-)

Thanks for urllib!
